### PR TITLE
Add wagtail aa1 to master

### LINKED ISSUE #1: There should be a easy to use CMS
**Describe the feature or change you'd like**
I would like to edit my Website without feeling visually attacked like with django CMS.

**Additional context**
Wagtail would be a perfect solution.
image

### LINKED ISSUE #2: There should be a customisable user model
**Describe the feature or change you'd like**
A custom user model would greatly improve the versatility of this template.

**Additional context**
image

### LINKED ISSUE #3: There should be a customisable image model
**Describe the feature or change you'd like**
A custom image model would greatly improve the versatility of this template.

**Additional context**
image

### LINKED ISSUE #4: There should be a customisable document model
**Describe the feature or change you'd like**
A custom document model would greatly improve the versatility of this template.

**Additional context**
shutterstock_296793953 0

### LINKED ISSUE #5: There should be a app that contains tools and snippets
**Describe the feature or change you'd like**
I think it would be a great Idea to add a repository with regularity used tools.

**Additional context**
image

### LINKED ISSUE #6: There should be a app for navbars
**Describe the feature or change you'd like**
A magical app that helps taming the beast called navbars.

**Additional context**
A rare image of the uppermentiond monstrosity.
image

### LINKED ISSUE #7: There should be a field and a block that can store colors
**Describe the feature or change you'd like**
It would be very pleasant if there would be a way to define the color of an object via CMS.

**Additional context**
image

### LINKED ISSUE #8: There should be a way to communicate with a PostgreSQL
**Describe the feature or change you'd like**
A PostgreSQL database adapter should be installed.
I think psycopg2 would be a good choice.

**Additional context**
https://pypi.org/project/psycopg2/

### LINKED ISSUE #9: There should be a app that helps with atomic design
**Describe the feature or change you'd like**
A  app that helps to follow the Atomic design methodology.

**Additional context**

### PR DESCRIPTION
**Pull request by kleberbaum**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently there is no easy option to edit pages. #1 
- Currently the default user model is used. #2 
- Currently the default image model is used. #3 
- Currently the default image model is used. #4
- Currently there is app for global tools and snippets. #5
- Currently there is no app for navbars. #6
- Currently there is no way of storing colors in a Page. #7
- Currently there is no PostgreSQL database adapter installed. #8  
- Currently there there is no structure for atomic design. #9 
- Currently some requirements are outdated.

**What is the new behavior (if this is a feature change)?**
- Now there is a easy option to edit pages. Closes #1 
- Now a custom user model is used. Closes #2 
- Now a custom image model is used. Closes #3 
- Now a custom image model is used. Closes #4
- Now there is app for global tools and snippets. Closes #5
- Now there is a app for navbars. Closes #6
- Now there is a way of storing colors in a Page. Closes #7
- Now a PostgreSQL database adapter is installed. Closes #8  
- Now there is no structure for atomic design. Closes #9 
- Now some requirements are more up to date.

**Other information**:
- Ref: https://github.com/snek-at/tonic
- Ref: https://pypi.org/project/django-cors-headers/
- Ref: https://pypi.org/project/psycopg2/
- Ref: https://pypi.org/project/django-pattern-library/
- Ref: https://github.com/helsenorge/helsenorgelab.no

### CODE REVIEW

**Review by schettn — CHANGES_REQUESTED** [review]
- We have to migrate to our new licence tag.
- Also the code doesn't comply with [Black](https://github.com/psf/black) standard.

**Review by kleberbaum — COMMENTED** [review]
Should be ok to merge.

File: esite/core/apps.py [hunk 1]
```diff
@@ -3,7 +3,8 @@
 
 # This defines the name of the app.
 class CoreConfig(AppConfig):
-    name = 'core'
+    name = "core"
+
 
 # SPDX-License-Identifier: (EUPL-1.2)
 # Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/core/apps.py]: Change to Simon Prast

File: esite/documents/models.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+from wagtail.documents.models import Document as WagtailDocument
+
+
+class CustomDocument(AbstractDocument):
+    description = models.TextField(max_length=255, blank=True, null=True)
+    admin_form_fields = WagtailDocument.admin_form_fields + ("description",)
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/documents/models.py]: Change to Simon Prast.

File: esite/home/apps.py [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
 
-
 # This defines the name of the app.
 class HomeConfig(AppConfig):
-    name = 'home'
+    name = "home"
+
 
 # SPDX-License-Identifier: (EUPL-1.2)
 # Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/home/apps.py]: Change to Simon Prast

File: esite/images/models.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    image = models.ForeignKey(
+        "CustomImage", related_name="renditions", on_delete=models.CASCADE
+    )
+
+    class Meta:
+        unique_together = (("image", "filter_spec", "focal_point_key"),)
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/images/models.py]: Change to Simon Prast

File: esite/navigation/models.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    panels = [
+        StreamFieldPanel("primary_navigation"),
+        StreamFieldPanel("footer_links"),
+        FieldPanel("footer_bottom_text"),
+    ]
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/navigation/models.py]: Change to Simon Prast

File: esite/navigation/templatetags/navigation_tags.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+# Footer nav snippets
+"""
+@register.inclusion_tag('patterns/molecules/navigation/footernav.html', takes_context=True)
+def footernav(context):
+    request = context['request']
+    return {
+        'footernav': NavigationSettings.for_site(request.site).footer_navigation,
+        'request': request,
+    }
+"""
```
Comment (pinterid) [inline esite/navigation/templatetags/navigation_tags.py]:
```suggestion
```

File: esite/navigation/templatetags/navigation_tags.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    return {
+        "footerbottomtext": NavigationSettings.for_site(
+            request.site
+        ).footer_bottom_text,
+        "request": request,
+    }
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/navigation/templatetags/navigation_tags.py]: Change to Simon Prast

File: esite/settings/base.py [hunk 8]
```diff
@@ -2,173 +2,268 @@
...
-CORS_ORIGIN_ALLOW_ALL = True
-
-#> URL configuration
-# A string representing the full Python import path to your root URLconf.
-# See https://docs.djangoproject.com/en/2.2/ref/settings/#root-urlconf
-ROOT_URLCONF = 'esite.urls'
+ROOT_URLCONF = "esite.urls"
 
-#> Template definition
+# > Template Configuration
```
Comment (pinterid) [inline esite/settings/base.py:93]: ?

File: esite/settings/base.py [hunk 9]
```diff
@@ -2,173 +2,268 @@
...
-MEDIA_URL = '/media/'
+# > Styleguide
+PATTERN_LIBRARY_ENABLED = "true"
+PATTERN_LIBRARY_TEMPLATE_DIR = "templates"
 
+# > System Checks
+# Wagtail forms not used so silence captcha warning
+SILENCED_SYSTEM_CHECKS = ["captcha.recaptcha_test_key_error"]
 # SPDX-License-Identifier: (EUPL-1.2)
 # Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/settings/base.py]: Change to Simon Prast

File: esite/settings/development.py [hunk 10]
```diff
@@ -2,37 +2,42 @@
...
+# See https://docs.djangoproject.com/en/stable/topics/email/#console-backend
+EMAIL_BACKEND = "django.core.mail.backends.console.EmailBackend"
+
+# > Set Base_Url
+# Set the base url, needed to acces wagtail.
+# See https://docs.wagtail.io/en/v0.8.10/howto/settings.html
+BASE_URL = "http://localhost:8000"
 
 # SPDX-License-Identifier: (EUPL-1.2)
 # Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/settings/development.py]: Change to Simon Prast

File: esite/settings/production.py [hunk 11]
```diff
@@ -19,47 +19,97 @@
...
+# > Recaptcha
+# These settings are required for the captcha challange to work.
+# https://github.com/springload/wagtail-django-recaptcha
+if "RECAPTCHA_PUBLIC_KEY" in env and "RECAPTCHA_PRIVATE_KEY" in env:
+    NOCAPTCHA = True
+    RECAPTCHA_PUBLIC_KEY = env["RECAPTCHA_PUBLIC_KEY"]
+    RECAPTCHA_PRIVATE_KEY = env["RECAPTCHA_PRIVATE_KEY"]
 
 # SPDX-License-Identifier: (EUPL-1.2)
 # Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/settings/production.py]: Change to Simon Prast

File: esite/urls.py [hunk 12]
```diff
@@ -13,12 +13,94 @@
...
+        path("", include(wagtail_urls)),
+    ]
+)
+
+# Error handlers
+handler404 = "esite.utils.views.page_not_found"
+handler500 = "esite.utils.views.server_error"
+
 # SPDX-License-Identifier: (EUPL-1.2)
 # Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/urls.py]: Change to Simon Prast

File: esite/user/admin.py [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+        UserAdmin,
+        # CustomerAdmin,
+        # RegistrationAdmin
+    )
+
+
+# modeladmin_register(CustomerAdminB)
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/user/admin.py]: Change to Simon Prast

File: esite/user/apps.py [hunk 14]
```diff
@@ -0,0 +1,9 @@
+from django.apps import AppConfig
+
+
+class UsersConfig(AppConfig):
+    name = "user"
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/user/apps.py]: Change to Simon Prast

File: esite/user/blocks.py [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+        else:
+            return ""
+
+    class Meta:
+        icon = "user"
+        name = "user"
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/user/blocks.py]: Change to Simon Prast

File: esite/user/models.py [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+        FieldPanel("is_customer"),
+        FieldPanel("registration_data"),
+    ]
+
+    def __str__(self):
+        return self.username
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/user/models.py]: Change to Simon Prast

File: esite/user/schema.py [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+
+    @login_required
+    def resolve_me(self, info, **_kwargs):
+        user = info.context.user
+
+        return user
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/user/schema.py]: Change to Simon Prast

File: esite/user/tests.py [hunk 18]
```diff
@@ -0,0 +1,6 @@
+from django.test import TestCase
+
+# Create your user related tests here.
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/user/tests.py]: Change to Simon Prast

File: esite/user/views.py [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+    model = get_user_model()
+    page_title = _("Choose a User")
+    per_page = 10
+    form_class = UserForm
+
+
+# Create your views here.
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/user/views.py]: Change to Simon Prast

File: esite/user/wagtail_hooks.py [hunk 20]
```diff
@@ -0,0 +1,15 @@
...
+from .views import UserChooserViewSet
+
+
+@hooks.register("register_admin_viewset")
+def register_user_chooser_viewset():
+    return UserChooserViewSet("user_chooser", url_prefix="user-chooser")
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/user/wagtail_hooks.py]: Change to Simon Prast

File: esite/user/widgets.py [hunk 21]
```diff
@@ -0,0 +1,16 @@
...
+
+
+class UserChooser(AdminChooser):
+    choose_one_text = _("Choose a User")
+    model = get_user_model()
+    choose_modal_url_name = "user_chooser:choose"
+
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline esite/user/widgets.py]: Change to Simon Prast

File: esite/utils/wagtail_hooks.py [hunk 22]
```diff
@@ -0,0 +1,27 @@
...
+    menu_label = "Taxonomies"
+    items = (
+        CategoryModelAdmin,
+        # PersonTypeModelAdmin,
+    )
+    menu_icon = 'tag'
+
+
+modeladmin_register(TaxonomiesModelAdminGroup)
+"""
```
Comment (pinterid) [inline esite/utils/wagtail_hooks.py:27]:
```suggestion
```

**Review by pinterid — CHANGES_REQUESTED** [review]
A few changes should be made

File: esite/settings/base.py [hunk 23]
```diff
@@ -2,173 +2,268 @@
...
 For more information on this file, see
-https://docs.djangoproject.com/en/2.2/topics/settings/
+https://docs.djangoproject.com/en/stable/topics/settings/
 
 For the full list of settings and their values, see
-https://docs.djangoproject.com/en/2.2/ref/settings/
+https://docs.djangoproject.com/en/stable/ref/settings/
 """
 
 import os
```
Comment (kleberbaum) [inline esite/settings/base.py:14]: The env variable has to be assigned in the base settings.
```suggestion
env = os.environ.copy()

```

**Review by kleberbaum — COMMENTED** [review]
Should be ok to merge now.

**Review by schettn — APPROVED** [review]
LGTM! :)

**Review by pinterid — APPROVED** [review]
Looks ok to merge.

**Pull request merged**